### ISSUE TITLE
Rule idea: forbid unnamespaced module usages

### ISSUE THREAD

**Issue by stof** [stylelint-scss/stylelint-scss#565]
Unnamespaced modules are exposing their members in the current scope without namespace, but those members are not copied. So assigning a value in a variable will assign it inside the imported module, allowing to bypass the rule suggested in #556.
I'd like to suggest a rule forbidding such unnamespaced imports, that can be used by projects that want that safety.

Invalid:

```scss
@use "foo" as *;
```

Valid:
```scss
@use "foo";
```

```scss
@use "foo" as bar;
```

Note that the official Sass documentation [explicitly recommends](https://sass-lang.com/documentation/at-rules/use#choosing-a-namespace) to use `as *` only for your own stylesheets rather than for third-party ones due to the risk of variable conflicts (where a value that was meant to be assigned locally suddenly modifies the external module).
So maybe this rule should have a option to switch between 2 modes:
- forbidding all non-namespaced usages
- forbidding non-namespaced usages only for non-relative imports
But this might be hard to implement, because relative imports in Sass are not required to start with `./` so looking at the current file alone is not enough to recognize relative imports.

For my own use case, I'm intereseted into forbidding all usages so I don't mind if the configuration option cannot be implemented.

**Comment by kristerkari** [comment]
Thanks for the detailed explanation. I think that it could be good rule to have.
